File: Annadata/app/src/main/java/com/mananhirak/annadata/notify/FireNotify.java
package com.mananhirak.annadata.notify;

import android.util.Log;

import androidx.annotation.NonNull;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;
import com.mananhirak.annadata.idtag.FireTag;
import com.mananhirak.annadata.idtag.Mtag;

import org.jetbrains.annotations.NotNull;

import java.util.Map;

public class FireNotify extends FirebaseMessagingService {
    @Override
    public void onMessageReceived(@NonNull @NotNull RemoteMessage remoteMessage) {
       super.onMessageReceived(remoteMessage);

        if(remoteMessage.getData().size()>0){

            Map<String,String> m=remoteMessage.getData();

            String title = m.get("title");
            String message = m.get("message");
            Log.d(Mtag.Tag, "onMessageReceived: "+m.get("id"));

            NotifyBuild build = new NotifyBuild(getApplicationContext());

            build.BuildNotification(title,message,m.get("id"));


        }else{
            Log.d(Mtag.Tag, "onMessageReceivertrtd: "+remoteMessage);
        }

    }
}

File: Annadata/app/src/main/java/com/manan/annadata/notify/FireNotify.kt
package com.manan.annadata.notify

import android.os.Build
import android.util.Log
import androidx.annotation.RequiresApi
import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import com.manan.annadata.idtag.Mtag

class FireNotify : FirebaseMessagingService() {
    @RequiresApi(Build.VERSION_CODES.S)
    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        super.onMessageReceived(remoteMessage)
        if (remoteMessage.data.size > 0) {
            val m = remoteMessage.data
            val title = m["title"]
            val message = m["message"]
            Log.d(Mtag.Tag, "onMessageReceived: " + m["id"])
            val build = NotifyBuild(applicationContext)
            build.BuildNotification(title, message, m["id"])
        } else {
            Log.d(Mtag.Tag, "onMessageReceivertrtd: $remoteMessage")
        }
    }
}